mesonbuild/interpreter/compiler.py

```diff
@@ -295,6 +295,11 @@ def alignment_method(self, args: T.Tuple[str], kwargs: 'AlignmentKw') -> int:
     @typed_pos_args('compiler.run', (str, mesonlib.File))
     @typed_kwargs('compiler.run', *_COMPILES_KWS)
     def run_method(self, args: T.Tuple['mesonlib.FileOrString'], kwargs: 'CompileKW') -> 'RunResult':
+        from ..compilers.d import DCompiler
+        from ..compilers.mixins.clike import CLikeCompiler
+        if not isinstance(self.compiler, (CLikeCompiler, DCompiler)):
```

Comment:
I thought the general move was away from this kind of import and isinstance checking. could we just do:
```python
if self.compiler.language not in {'d', 'c', 'cpp', 'objc', 'objcpp'}:
     FeatureNew.single_use(...)
```

Reply:
Good point, changed.